File: test/crc32.test.js
const { runTest } = require("./test-helpers.js");
const test = require("tape");

test("CRC32 basic functionality", async (t) => {
  const expectedOutput = `Test passed`;

  const result = await runTest("CRC32Test", expectedOutput, t);

  t.equal(
    result.output.trim(),
    expectedOutput.trim(),
    "CRC32 should calculate correct checksum for 'Hello, World!'"
  );
  t.ok(result.success, "Test should complete successfully");
  t.end();
});

test("CRC32 step by step operations", async (t) => {
  const expectedOutput = `CRC32 created\nCRC32 updated\nValue: 4157704578`;

  const result = await runTest("CRC32GetValueTest", expectedOutput, t);

  t.equal(
    result.output.trim(),
    expectedOutput.trim(),
    "CRC32 should work with step-by-step operations"
  );
  t.ok(result.success, "Test should complete successfully");
  t.end();
});

test("CRC32 reset functionality", async (t) => {
  const expectedOutput = `CRC32 created\nCRC32 reset`;

  const result = await runTest("CRC32SimpleTest", expectedOutput, t);

  t.equal(
    result.output.trim(),
    expectedOutput.trim(),
    "CRC32 should support reset operations"
  );
  t.ok(result.success, "Test should complete successfully");
  t.end();
});

File: sources/CRC32Test.java
import java.util.zip.CRC32;

public class CRC32Test {
    public static void main(String[] args) {
        // Test reset()
        CRC32 crc = new CRC32();
        System.out.println("CRC32 created");
        crc.reset();
        System.out.println("CRC32 reset");

        // Test update() and getValue()
        String testString = "Hello, World!";
        byte[] bytes = testString.getBytes();
        
        crc.update(bytes, 0, bytes.length);
        System.out.println("CRC32 updated");

        long value = crc.getValue();
        System.out.println("Value: " + value);
        
        // The expected CRC32 value for "Hello, World!" is 3964322768
        if (value == 3964322768L) {
            System.out.println("Test passed");
        } else {
            System.out.println("Test failed: expected 3964322768, got " + value);
        }
    }
}